build/includes/sdk.mk

```diff
@@ -251,3 +251,33 @@ test-gen-all-sdk-grpc:
 		echo "$$diff_output_test_sdk"; \
 		exit 1; \
 	fi
+
+# Ensure examples images are bumped when there is any change in the scripts
+check-makefile-version-increment:
+	@echo "Current directory: $$(pwd)"
+	@echo "Checking for script changes in 'examples' subdirectories..."
+	@cd $(CURDIR)/..; \
+	changed_dirs=$$(git diff --name-only HEAD | grep -E "examples/.*/(main\.go|Dockerfile|Dockerfile\.windows|go\.mod|go\.sum)$$" | xargs -n1 -r dirname | sort -u); \
```

Comment:
Ow my head hurts 😄 can we get some descriptive comments in here to work out what this is all checking?

Reply:
Sorry, I agree the script is confusing. Could you please share a better approach? I don't think this bash script will work well.

Reply:
> changed_dirs=$$(git diff --name-only HEAD | grep -E "examples/.*/(main\.go|Dockerfile|Dockerfile\.windows|go\.mod|go\.sum)$$" | xargs -n1 -r dirname | sort -u); \

This will find all the changed files under `examples` subdirectories that match the given patterns (main.go, Dockerfile, Dockerfile.windows, go.mod, go.sum), after that it will extract the directories containing these files, remove duplicates, and store them in `changed_dirs`.

Reply:
> Sorry, I agree the script is confusing. Could you please share a better approach? I don't think this bash script will work well.

Go script? You've created some really nice ones in the past. Probably easier too!

https://pkg.go.dev/github.com/go-git/go-git/v5 looks like the dominant Go Git library.

On failure, you could `log.Fatalf(...)` and CI would fail, since the exit code of the script would be non-zero.

> This will find all the changed files under examples subdirectories that match the given patterns (main.go, Dockerfile, Dockerfile.windows, go.mod, go.sum)

Since each of the examples use different languages, I'm wondering if we want to exclude things we know we don't care about (*.md, *.yaml, OWNERS, .gitignore) rather than an include list - as the number of examples and language will expand over time.

Reply:
Thanks for your feedback. I will implement a Go script👍

Reply:
Update: The Go script implementation is in progress. Looks like bash script is simpler than the Go script. Will push the commit once completed.

Reply:
> Looks like bash script is simpler than the Go script

I'm hoping it's easier to understand though 😄 `grep -oP '^[^#]*version\s*:=\s*\K[^ ]+'` - that scared me 😨 😀

Reply:
I have implemented a Go script but ignoring comments from the files is the challenging part for me. https://gist.github.com/Kalaiselvi84/cadc597d7fa2c229ecf2a992ddbd4b01 

I think it is infeasible for me, giving up. I will close this PR. Thanks for your help and support.🙏